docs/api/MockAgent.md

```diff
@@ -0,0 +1,388 @@
+
+```js
+'use strict'
+const { MockAgent } = require('undici')
+
+const mockAgent = new MockAgent()
+setGlobalAgent(mockAgent)
+
+const mockPool = mockAgent.get(new RegExp('http://localhost:3000'))
+mockPool.intercept({
+  path: '/foo',
+  method: 'GET',
+}).reply(200, 'foo')
+
+const {
+  statusCode,
+  body
+} = await request('http://localhost:3000/foo')
+
+console.log('response received', statusCode) // 200
+
+for await (const data of body) {
+  console.log('data', data) // 'foo'
+}
+```
+
+#### Example - Mocked request with origin function
+
+```js
+'use strict'
+const { MockAgent } = require('undici')
+
+const mockAgent = new MockAgent()
+setGlobalAgent(mockAgent)
+
+const mockPool = mockAgent.get((origin) => 'http://localhost:3000' === origin))
+mockPool.intercept({
+  path: '/foo',
+  method: 'GET',
+}).reply(200, 'foo')
+
+const {
+  statusCode,
+  body
+} = await request('http://localhost:3000/foo')
+
+console.log('response received', statusCode) // 200
+
+for await (const data of body) {
+  console.log('data', data) // 'foo'
+}
+```
+
+### `MockAgent.close()`
+
+Closes the mock agent and waits for registered mock pools and clients to also close before resolving.
+
+Returns: `Promise<void>`
+
+#### Example - clean up after tests are complete
+
+```js
+'use strict'
+const { MockAgent } = require('undici')
+
+const mockAgent = new MockAgent()
+
+await mockAgent.close()
+```
+
+### `MockAgent.deactivate()`
+
+This method disables mocking in MockAgent.
+
+Returns: `void`
+
+#### Example - Deactivate Mocking
+
+```js
+'use strict'
+const { MockAgent } = require('undici')
+
+const mockAgent = new MockAgent()
+
+mockAgent.deactivate()
+```
+
+### `MockAgent.activate()`
+
+This method enables mocking in a MockAgent instance. When instantiated, a MockAgent is automatically activated. Therefore, this method is only effective after `MockAgent.deactivate` has been called.
+
+Returns: `void`
+
+#### Example - Activate Mocking
+
+```js
+'use strict'
+const { MockAgent } = require('undici')
+
+const mockAgent = new MockAgent()
+
+mockAgent.deactivate()
+// No mocking will occur
+
+// Later
+mockAgent.activate()
+```
+
+### `MockAgent.enableNetConnect([host])`
+
+When requests are not matched in a MockAgent intercept, a real HTTP request is attempted. We can control this further through the use of `enableNetConnect`. This is achieved by defining host matchers so only matching requests will be attempted.
+
+When using a string, it should only include the **hostname and optionally, the port**. In addition, calling this method multiple times with a string will allow all HTTP requests that match these values.
+
+Arguments:
+
+* **host** `string | RegExp | (value) => boolean` - (optional)
+
+Returns: `void`
+
+#### Example - Allow all non-matching urls to be dispatched in a real HTTP request
+
+```js
+'use strict'
+const { MockAgent } = require('undici')
+
+const mockAgent = new MockAgent()
+
+mockAgent.enableNetConnect()
+
```

Comment:
Shouldn't we have `setGloblAgent` here and in the following examples?

Reply:
Good spot - I'll fix those now!